crypto/bio/bss_mem.c

```diff
@@ -157,37 +168,60 @@ static int secmem_new(BIO *bi)
 
 static int mem_free(BIO *a)
 {
+    return (mem_buf_free(a, 1));
+}
+
+static int mem_buf_free(BIO *a, int free_all)
+{
     if (a == NULL)
         return (0);
     if (a->shutdown) {
         if ((a->init) && (a->ptr != NULL)) {
             BUF_MEM *b;
-            b = (BUF_MEM *)a->ptr;
-            if (a->flags & BIO_FLAGS_MEM_RDONLY)
-                b->data = NULL;
-            BUF_MEM_free(b);
+            BIO_BUF_MEM *bb;
+            if((bb = (BIO_BUF_MEM *)a->ptr) != NULL) {
```

Comment:
put the "bb" assignment on its declaration line.  put a blank line after the local vars.  then the if test becomes "bb != NULL"